src/Hosting/TestHost/src/WebSocketClient.cs

```diff
@@ -59,6 +59,14 @@ public async Task<WebSocket> ConnectAsync(Uri uri, CancellationToken cancellatio
                 scheme = (scheme == "ws") ? "http" : scheme;
                 scheme = (scheme == "wss") ? "https" : scheme;
                 request.Scheme = scheme;
+                if (!request.Host.HasValue)
+                {
+                    request.Host = HostString.FromUriComponent(uri);
```

Comment:
Check if the uri.IsDefaultPort right here so you can avoid the double assignment?

Reply:
I've got my inspiration from [here](https://github.com/aspnet/AspNetCore/blob/master/src/Hosting/TestHost/src/ClientHandler.cs#L83)